packages/react-components/src/components/MIMap/MapControls/MapControls.jsx

```diff
@@ -0,0 +1,125 @@
+    const isDesktop = useIsDesktop();
+    const floorSelectorRef = useRef(null);
+    const positionButtonRef = useRef(null);
+
+    // Define portal elements as constants
+    const venueSelectorPortal = <div key="venue-selector" className="venue-selector-portal" />;
+    const floorSelectorPortal = <div key="floor-selector" className="floor-selector-portal" />;
+    const myPositionPortal = <div key="my-position" className="my-position-element-portal" />;
+    const viewModeSwitchPortal = <div key="viewmode-switch" className="viewmode-switch-portal" />;
+
+    // Create and configure web components
+    // This useEffect will run when the component mounts and when the mapsIndoorsInstance or mapInstance changes.
+    // It will create the web components and set their properties.
+    // The web components are created only once and reused when the component re-renders.
+    useEffect(() => {
+        if (!mapsIndoorsInstance || !mapInstance) return;
+
+        // Create the web components if they don't exist
+        if (!floorSelectorRef.current) {
+            const floorSelector = document.createElement('mi-floor-selector');
+            floorSelectorRef.current = floorSelector;
+        }
+
+        // Create the position button if it doesn't exist
+        if (!positionButtonRef.current) {
+            const positionButton = document.createElement('mi-my-position');
+            positionButtonRef.current = positionButton;
+        }
+
+        // Update properties of the floor selector and position button
+        floorSelectorRef.current.mapsindoors = mapsIndoorsInstance;
+        positionButtonRef.current.mapsindoors = mapsIndoorsInstance;
+
+        if (brandingColor) {
+            floorSelectorRef.current.primaryColor = brandingColor;
+        }
+
+        // Setup position control
+        if (onPositionControl && positionButtonRef.current) {
+            onPositionControl(positionButtonRef.current);
+        }
+
+    }, [mapType, mapsIndoorsInstance, mapInstance, onPositionControl, brandingColor]);
+
+    // Handle layout changes and element movement, this handles moving the elements to the correct DOM location based on the layout
+    // and ensures that the elements are not duplicated in the DOM.
+    // This is important for performance and to avoid issues with the web components.
+    // The useEffect will run when the component mounts and when the layout changes (isDesktop changes).
+    useEffect(() => {
+        if (!floorSelectorRef.current || !positionButtonRef.current) return;
+
+        // Function to move elements to the target container
+        // This function will check if the element is already in the target container and move it if not.
+        const moveElementToTarget = (element, targetClass) => {
+            const target = document.querySelector(`.${targetClass}`);
+            if (target && !target.contains(element)) {
+                element.parentElement?.removeChild(element);
```

Comment:
Could we just use remove()? 

https://developer.mozilla.org/en-US/docs/Web/API/Element/remove